client/client.go

```diff
@@ -259,23 +264,38 @@ func (client *Client) Hijack(f func(*http.Request) (*http.Response, error)) {
 	client.doer = hijacked{f}
 }
 
+type doFlags uint
+
+const (
+	doDefaults  doFlags = 0
+	doNoTimeout doFlags = 1 << iota
```

Comment:
we should use a struct with bools as for everything else

Reply:
Refactored to use a struct